src/getMigrateDashboards.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  DashboardState as DashboardState50,
+  ContentRecord,
+} from "@activeviam/activeui-sdk-5.0";
+import { DashboardState as DashboardState51 } from "@activeviam/activeui-sdk-5.1";
+import _omit from "lodash/omit";
+import {
+  ErrorReport,
+  MigrateFromVersions,
+  OutcomeCounters,
+} from "./migration.types";
+import { _addErrorToReport } from "./_addErrorToReport";
+import { _getMapOfFolderIds } from "./_getMapOfFolderIds";
+import { _serializeError } from "./_serializeError";
+
+type MigrateDashboardCallback<fromVersion extends MigrateFromVersions> = (
+  dashboardState: fromVersion extends "5.0" ? DashboardState50 : any,
+) => fromVersion extends "5.0" ? DashboardState51 : any;
+
+/**
+ * Handles the generic logic to migrate dashboards, specifically the errors.
+ * `cb` is a generic callback that is meant to handle dashboards migration from one specific version to another.
+ * Mutates `migratedUIFolder`, `dashboards`, `counters` and `errorReport`.
+ */
```

Comment:
If we go for: "`getMigrateDashboards` returns a function", then I would suggest:
```suggestion
/**
 * Returns a function which can be called to migrate the content of each dashboard, according to the logic defined in its `callback` argument.
 *
 * The logic of traversing each file in `/ui/dashboards` and the error handling is encapsulated to make it easier for the caller.
 *
 * Mutates `contentServer`, `counters` and `errorReport`.
 */
```

Reply:
By the way, now that I think of it, it doesn't mutate those variables, the mutation is done by the migration function that is returned. I'll update the comment